modal/billing.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Modal Labs 2025
+from datetime import datetime, timezone
+from decimal import Decimal
+from typing import Any, Optional, Union
+
+from modal_proto import api_pb2
+
+from ._utils.async_utils import synchronizer
+from .client import _Client
+from .exception import InvalidError
+
+__all__ = ["workspace_billing_report"]
+
+
+@synchronizer.create_blocking
+async def workspace_billing_report(
+    *,
+    start: Union[datetime, str],  # Start of the report, inclusive
+    end: Optional[Union[datetime, str]] = None,  # End of the report, exclusive
+    resolution: str = "d",  # Resolution, e.g. "d" for daily or "h" for hourly
```

Comment:
Could maybe be
```suggestion
    resolution: Literal["d", "h"] = "d",  # Resolution, e.g. "d" for daily or "h" for hourly
```
?

Reply:
Currently, the server defines what we accept here. Maybe it would be better to define that in the client and use a literal type so it's more obvious. It would be annoying if we added support for more resolutions in the future and required users to upgrade their client to get access, since all the client needs to do is pass this parameter up to the server. OTOH, it's unlikely that we'll support something like minute resolution in this interface, so perhaps we don't need to design for future expansion.